common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/param/FormProcessorCreator.java

```diff
@@ -50,20 +52,33 @@ public Object getValue(HttpServletRequest request) {
       }
 
       if (targetType.isContainerType()) {
-        return convertValue(request.getParameterValues(paramPath), targetType);
+        Object values = request.getParameterValues(paramPath);
+        //Even if the paramPath does not exist, it won't be null at now, may be optimized in the future
+        if (values == null) {
+          values = checkRequiredAndDefaultValue(values);
+        }
+        return convertValue(values, targetType);
       }
 
       Object value = request.getParameter(paramPath);
       if (value == null) {
-        Object defaultValue = getDefaultValue();
-        if (defaultValue != null) {
-          value = defaultValue;
-        }
+        value = checkRequiredAndDefaultValue(value);
       }
 
       return convertValue(value, targetType);
     }
 
+    private Object checkRequiredAndDefaultValue(Object values) {
```

Comment:
values is always null, no need to pass this parameter

Reply:
done